server-plugins/activity-resources/src/index.ts

```diff
@@ -349,9 +349,18 @@ export async function generateDocUpdateMessages (
         let doc = objectCache?.docs?.get(tx.objectId)
         if (doc === undefined) {
           doc = (await control.findAll(ctx, tx.objectClass, { _id: tx.objectId }, { limit: 1 }))[0]
-          objectCache?.docs?.set(tx.objectId, doc)
+        }
+        if (doc === undefined) {
+          const createTx =
+            (await control.findAll(ctx, core.class.TxCreateDoc, { objectId: tx.objectId }, { limit: 1 }))[0] ??
```

Comment:
Could we check by AttachedDoc and find only a once?

Reply:
Done.